homeassistant/components/rotel/media_player.py

```diff
@@ -0,0 +1,493 @@
+        self._volume = None
+        self.set_source_lists(source_aliases)
+        self._mute_on = None
+        self._party_mode_on = None
+        self._info = None
+        self._icons = None
+        self._speaker_icons = None
+        self._state_icons = None
+        self._input_icons = None
+        self._sound_mode_icons = None
+        self._misc_icons = None
+        self._triggers = None
+
+    @property
+    def should_poll(self) -> bool:
+        """Do not poll because this entity pushes its state to HA."""
+        return False
+
+    async def open_connection(self):
+        """Open a connection to the device."""
+        from rsp1570serial.connection import RotelAmpConn
+        conn = RotelAmpConn(self._device)
+        try:
+            await conn.open()
+        except Exception:
+            _LOGGER.error("Could not open connection", exc_info=True)
+            raise
+        self._conn = conn
+
+    def close_connection(self):
+        """Close the connection to the device."""
+        if self._conn is not None:
+            self._conn.close()
+
+    def start_read_messages(self, hass):
+        """Create a task to start reading messages."""
+        self._read_messages_task = hass.loop.create_task(
+            self.async_read_messages()
+        )
+
+    async def cancel_read_messages(self):
+        """Cancel the _read_messages_task."""
+        if self._read_messages_task is not None:
+            _LOGGER.info("Cancelling read_messages task.  Done was: %r.",
+                         self._read_messages_task.done())
+            if not self._read_messages_task.done():
+                self._read_messages_task.cancel()
+                await self._read_messages_task
+            else:
+                ex = self._read_messages_task.exception()
+                if ex is not None:
+                    _LOGGER.error(
+                        "Read messages task contained an exception.",
+                        exc_info=ex)
+            self._read_messages_task = None
+
+    async def async_read_messages(self):
+        """
+        Read messages published by the device.
+
+        Read messages published by the device and use them to maintain the
+        state of this object.
+        Schedule a DISPLAY_REFRESH command before we start reading.
+        If the device is already on then this is a null command that will
+        simply trigger a feedback message that will sync the state of this
+        object with the physical device.
+        """
+        _LOGGER.info("Message reader started for %s.", self.name)
+        try:
+            await self.send_command('DISPLAY_REFRESH')
+            async for message in self._conn.read_messages():
+                _LOGGER.debug("Message received by %s.", self.name)
+                self.handle_message(message)
+        except asyncio.CancelledError:
+            _LOGGER.info("Message reader cancelled for %s", self.name)
+        except Exception:
+            _LOGGER.error((
```

Comment:
Don't log and raise -- this is already being logged by the code that cancels the task.